Public/Src/Cache/DistributedCache.Host/Service/Internal/CacheServerFactory.cs

```diff
@@ -0,0 +1,238 @@
+            {
+                Func<AbsolutePath, ICache> cacheFactory = path => {
+                    return new OneLevelCache(
+                        contentStoreFunc: () => contentStoreFactory(path),
+                        memoizationStoreFunc: () => {
+                            return new RocksDbMemoizationStore(_logger, SystemClock.Instance, new RocksDbMemoizationStoreConfiguration()
+                            {
+                                Database = new RocksDbContentLocationDatabaseConfiguration(path / "RocksDbMemoizationStore") {
+                                    MetadataGarbageCollectionEnabled = true,
+                                },
+                            });
+                        },
+                        Guid.NewGuid(),
+                        passContentToMemoization: true);
+                };
+
+                return new LocalCacheServer(
+                    _fileSystem,
+                    _logger,
+                    _arguments.Configuration.LocalCasSettings.ServiceSettings.ScenarioName,
+                    cacheFactory,
+                    localServerConfiguration);
+            }
+            else
+            {
+                return new LocalContentServer(
+                    _fileSystem,
+                    _logger,
+                    _arguments.Configuration.LocalCasSettings.ServiceSettings.ScenarioName,
+                    contentStoreFactory,
+                    localServerConfiguration);
+            }
+        }
+
+        private StartupShutdownBase CreateDistributedServer(LocalServerConfiguration localServerConfiguration)
+        {
+            var cacheConfig = _arguments.Configuration;
+
+            var hostInfo = _arguments.HostInfo;
+            _logger.Debug($"Creating on stamp id {hostInfo.StampId} with scenario {cacheConfig.LocalCasSettings.ServiceSettings.ScenarioName ?? string.Empty}");
+
+            var factory = new DistributedContentStoreFactory(
+                _arguments,
+                cacheConfig.DistributedContentSettings.GetRedisConnectionSecretNames(hostInfo.StampId));
+
+            Func<AbsolutePath, IContentStore> contentStoreFactory = path =>
+            {
+                var cacheSettingsByCacheName = cacheConfig.LocalCasSettings.CacheSettingsByCacheName;
+                var drivesWithContentStore = new Dictionary<string, IContentStore>(StringComparer.OrdinalIgnoreCase);
+
+                foreach (var settings in cacheSettingsByCacheName)
+                {
+                    _logger.Debug($"Using [{settings.Key}]'s settings: {settings.Value}");
+
+                    var rootPath = cacheConfig.LocalCasSettings.GetCacheRootPathWithScenario(settings.Key);
+                    drivesWithContentStore[GetRoot(rootPath)] = factory.CreateContentStore(rootPath, replicationSettings: null);
+                }
+
+                return new MultiplexedContentStore(drivesWithContentStore, cacheConfig.LocalCasSettings.PreferredCacheDrive);

```

Comment:
In this case, we are multiplexing. Is it correct to still return a single memoization store?